packages/common/src/hooks/useHostContext.ts

```diff
@@ -0,0 +1,21 @@
+import create from 'zustand';
+import { LocaleKey, SupportedLocales } from '../intl/intlHelpers';
+
+type HostContext = {
+  locale: SupportedLocales;
+  titleKey?: LocaleKey;
+  setLocale: (locale: SupportedLocales) => void;
+  setTitleKey: (titleKey: LocaleKey) => void;
+};
+
+const localStorageKey = '@openmsupply-client/localisation/locale';
+
+export const useHostContext = create<HostContext>(set => ({
+  locale: (localStorage.getItem(localStorageKey) as SupportedLocales) ?? 'en',
```

Comment:
I wonder actually- what if we wrapped the provider in our own provider.. 



```
const TranslationContext.Provider = React.createContext(getFromLocalStorageOrSomething?)

const TranslationProvider = ({children}) => {
 const [locale, setLocale] = useLocalStorageState(getLocaleFromLocalStorage)
 return <TranslationContext.Provider><IntlProvider locale={locale}>{children}</IntlProvider> </TranslationContext.Provider>
}

const useChangeLocale = () => useContext(TranslationContext)
```

I dunno

Reply:
yup. I went on that journey and had too many problems, so wound it all back again.
I thought to expose the other helpers too.. so I could `const { t, setLocale, etc } = useTranslationContext()` but I think now that this wouldn't get past the review stage.

Still like the idea of wrapping the `IntlProvider` in our own context.